cellprofiler/modules/loadimages.py

```diff
@@ -3367,9 +3378,61 @@ def provide_image(self, image_set):
 
     def __provide_volume(self):
         pathname = url2pathname(self.get_url())
-
         if self.is_numpy_file():
             data = np.load(pathname)
+        elif self.is_omero3d_path():
+            url = self.get_url()
+            parsed_url = urlparse.urlparse(url)
+            query_params = urlparse.parse_qs(parsed_url.query)
+            zmin = int(query_params['zmin'][0])
+            zmax = int(query_params['zmax'][0])
+            width = int(query_params['width'][0])
+            height = int(query_params['height'][0])
+            stack = numpy.ndarray((zmax - zmin + 1, width, height), 'uint16')
+            if zmin != 0:
+                zmax = zmax - zmin
+                zmin = 0
+            for i in range(zmin, zmax):
+                path = regex.sub(r'tile/([0-9]*)/([0-9]*)/' ,r'/tile/\1/%s/'%i,
+                    parsed_url.path)
+                query = parsed_url.query.split('&zmin')[0]
+                url = urlparse.urlunparse((
+                    parsed_url.scheme,
+                    parsed_url.netloc,
+                    path,
+                    '',
+                    query,
+                    ''
+                ))
+                response = requests.get(url)
+                image_bytes = BytesIO(response.content)
+                image = Image.open(image_bytes)
+                stack[i - 1, :, :] = image
+            data = stack
+        elif self.is_zarr_path:
+            url = self.get_url()
+            parsed_url = urlparse.urlparse(url)
+            query_params = urlparse.parse_qs(parsed_url.query)
+            group = query_params['group'][0]
+            zmin = int(query_params['zmin'])[0]
```

Comment:
This is wrong and the array index should be inside the parenthesis, correct?

Reply:
yes sorry new commit coming for zarr url parsing.  oops re: array index 😅